Adding initial additions for a new modular armor system.

PGR-14 opened the pull request:
#### Summary

Content "Adds the start of some new modular survivor armor"

#### Purpose of change

I wanted to make a new armor system, based off of needs & haves, & not off of pre-mediated blueprints.

#### Describe the solution

Through ablative pockets, survivors can craft their own jackets & pants (For now), The actual worn item is essentially just the hollowed out core of the clothing article. You can add a backer, liner, & filler, along w/ a utility slot. Additionally, there's pockets for these as well, for instance: backers can have forearm protectors, an arm patch, & a mural.

#### Describe alternatives you've considered

Maybe something like gunmods? We still have vestigial parts from something like that. We did try making it using those, but it wasn't very promising.

#### Testing

Tested using debug menu, works fine, inserts do give armor & proper encumbrances, no errors.

#### Additional context

The multiple commits are due to me not giving proper credit. This is my first PR, so sorry if this is terrible.
DEFINITELY needs balancing, a lot of tuneups, & probably has some bugs I haven't seen.
I've also got an ungodly amount more content in back, I was advised not to overwhelm you guys, & see what criticisms & additions y'all got.
Give me *any* ideas, I like hearing them, & I've got both definite bias, & a singular viewpoint.

Commit:
Addin' the intitial stuff

Commit:
Undo, & redo

Commit:
Add the basic parts of the new armor system

Adds:
-Generic versions of each pocket insert, as well as examples for each.

Co-authored-by: BalthazarArgall <116303491+BalthazarArgall@users.noreply.github.com>
& Mordet <Djaid.adam.nassim@gmail.com>

PGR-14 committed:
Update data/json/items/armor/ModularArmor/Filling.json

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

PGR-14 committed:
Update data/json/items/armor/ModularArmor/Modu_ArmorInserts.json

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

PGR-14 committed:
Update data/json/items/armor/ModularArmor/Modu_ArmorInserts.json

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

PGR-14 committed:
Apply suggestions from code review

Add github suggestions

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

PGR-14 committed:
Apply suggestions from code review

Add more git suggestions

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

PGR-14 committed:
Apply suggestions from code review

Add the ones I missed

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

kevingranade commented:
The whole point of the recipe system is that articles of clothing are not modular in this way.  If it's going to have an armored or insulated "filling" is going to have a completely different design for each option.

In practice, from when we did have a system like this in the past,  the scaling of encumbrance, weight, and utility were impossible to keep in sync, leading to specific configurations being "best in the game" while also being nonsensical setups.

PGR-14 commented:
> The whole point of the recipe system is that articles of clothing are not modular in this way. If it's going to have an armored or insulated "filling" is going to have a completely different design for each option.
> 
> In practice, from when we did have a system like this in the past, the scaling of encumbrance, weight, and utility were impossible to keep in sync, leading to specific configurations being "best in the game" while also being nonsensical setups.

Filling being completely different designs is a fair point. It's less so a place to store fur, as it is available space for something in between the liner & backing.

That's a fair criticism, why don't we scale it w/ fab/tail skill, like survivor gear? 
~lvl 0-3 it's mostly just a regular 'ol jacket, maybe w/ some extra stuff tagged on, but not much else. Not much better than a regular jacket, but slightly preferable thanks to choosing what you want.
~4-7 is better, w/ available armor inserts to increase protection. Keep in mind that this isn't the whole thing, higher level & more technologically advanced additions are gonna be added.
~8-10 is great. Nomad & Hub-01 stuff is gonna be better, no doubt, but the survivor gear's one upside is lower relative encumbrance, & _mutant compatibility_.

Again, mostly framework. If you got any balance additions, _PLEASE_ tell me.

kevingranade commented:
My balance concern is that the thing you're purposing is impossible to balance.

PGR-14 commented:
> My balance concern is that the thing you're purposing is impossible to balance.

I mean, it's fair that the previous version might've been terrible, but maybe it was implemented badly, or maybe the earlier versions had less threats to counteract? We won't really know, lest we try.

kevingranade commented:
"Maybe something different will work" means you go see what didn't work before and why,  and propose something that addresses those issues,  not, "just let me try something you don't think will work".

PGR-14 commented:
> "Maybe something different will work" means you go see what didn't work before and why, and propose something that addresses those issues, not, "just let me try something you don't think will work".

Dude, at least work w/ me a little. Mind pointing me to where I could see where the previous one went wrong?

kevingranade commented:
I don't have the issues memorized so it would just be me searching for it instead of you searching for it, just search "padding" and some other related keywords.

PGR-14 commented:
> I don't have the issues memorized so it would just be me searching for it instead of you searching for it, just search "padding" and some other related keywords.

Thanks, isn't there some vestigual code still in the game?

PGR-14 commented:
> I don't have the issues memorized so it would just be me searching for it instead of you searching for it, just search "padding" and some other related keywords.

Looked through them, there wasn't much on actual balance, just adding & removing stuff. Can't find the PR that removes armor padding either. I'm probably just blind.

PGR-14 commented:
Me & Balthazar've decided to change it. Instead of making each individual part of each clothing, so that people can choose exactly what they want, we're gonna make a set of clothing for popular archetypes so that players can pick & choose for their own particular playstyle.

BalthazarArgall commented:
As @PGR-14 said, I'll help on the design side, an issue will be raised later to discuss the direction of things once we have a workable concept.

Maddremor commented:
You seem to be stuck on the notion of balance when the issue is having a system produce consistently sane outputs.

GuardianDll commented:
see #58035
